app/models/miq_queue.rb

```diff
@@ -163,12 +163,8 @@ def self.get(options = {})
     msgs.each do |msg|
       begin
         _log.info("#{MiqQueue.format_short_log_msg(msg)} previously timed out, retrying...") if msg.state == STATE_TIMEOUT
-        w = MiqWorker.server_scope.find_by(:pid => Process.pid)
-        if w.nil?
-          msg.update_attributes!(:state => STATE_DEQUEUE, :handler => MiqServer.my_server)
-        else
-          msg.update_attributes!(:state => STATE_DEQUEUE, :handler => w)
-        end
+        w = MiqWorker.server_scope.find_by(:pid => Process.pid) || MiqServer.my_server
```

Comment:
I like that better - BUT, after looking at `MiqQueue.get` I have another optimization here :)